src/core/syscalls/syscall_handler.rs

```diff
@@ -43,9 +44,20 @@ pub(crate) trait SyscallHandler {
 
     fn get_tx_info(
         &mut self,
-        vm: &VirtualMachine,
+        vm: &mut VirtualMachine,
```

Comment:
Why does this mutate the VM?

Reply:
It needs to write to the vm's memory on line 59

Reply:
Yes,
The `response.write_syscall_response(vm, syscall_ptr)` writes into de vm memory